### ISSUE TITLE
No way to capture mouse input

### ISSUE THREAD

**Issue by OneXDeveloper** [veeenu/hudhook#12]
Another modification I had to make to use this lib was reimplementing `is_capturing` on the `ImguiRenderLoop` trait. I wanted mouse clicks and text input to not pass through to the game when the imgui box is focused by the user. 

My exact implementation:

Trait impl:
```rust
fn is_capturing(&self, io: &imgui_dx12::imgui::Io) -> bool {
    io.want_capture_mouse
}
```

Default trait impl
```rust
fn is_capturing(&self, _: &imgui_dx12::imgui::Io) -> bool {
    false
}
```

Then near the bottom of `imgui_wnd_proc`:

```rust
if let Some(render_loop) = IMGUI_RENDER_LOOP.get() {
     if render_loop.is_capturing(imgui_renderer.ctx.io()) {
          return LRESULT(1);
     }
 }
 ```

Again, I don’t think this is the perfect implementation but it seems to work OK for my use case. If this could be implemented I believe I won’t have to vendor this lib anymore and can just use the one from git :)

**Comment by veeenu** [comment]
Would this work (in place of [this](https://github.com/veeenu/hudhook/blob/master/hudhook/src/hooks/dx12.rs#L271-L276))?

```rust
let want_capture = io.want_capture_mouse || io.want_capture_keyboard;
let wnd_proc = imgui_renderer.wnd_proc;
drop(imgui_renderer);

if want_capture {
    trace!("Leaving WndProc via capturing");
    LRESULT(1)
} else {
    trace!("Leaving WndProc via CallWindowProcW");
    CallWindowProcW(Some(wnd_proc), hwnd, umsg, WPARAM(wparam), LPARAM(lparam))
}
```

This should capture inputs in the way intended by imgui, and does not need to rely on augmenting the trait. Unfortunately it doesn't work for me as the games I built this for (DS3 & Elden Ring) seemingly don't even go through WndProc for gathering their input, but it is also not a problem for those so I can include it without introducing regressions.

It's already in `master`.

**Comment by OneXDeveloper** [comment]
Yes, I think that would solve my case. I was opting for the trait the method to make it a bit more flexible for other use cases but it may not be needed. Awesome! Thanks again
